### PR TITLE
Update gapi.client.classroom types using new type generator

### PR DESCRIPTION
**Pull request by declanvong**
I am still in the process of testing this in our own codebase, so I've just raised this as a draft PR for now.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

The existing generator scrapes the Google API Discovery Docs in order to generate the TS types (this is also how the client library itself is generated). However, it has been out of date for quite some time and various issues with the types it generates [have been raised](https://github.com/Bolisov/google-api-typings-generator/issues/). The maintainer doesn't seem to be active anymore.

I have [forked the generator](https://github.com/declanvong/google-api-typings-generator/) and updated it, and then regenerated all of the gapi types. This PR currently only contains the `gapi.client.classroom` types as a starter.

- [-] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by declanvong** [comment]
cc @joshuaobrien @ben-canva

**Comment by declanvong** [comment]
Tbh, not sure if there's much point in the massive tests since they're generated and the types are all generated anyway... but I've just updated the existing test generator to actually include data.

**Comment by orta** [comment]
This is still in draft, is it something you want to update?

**Comment by declanvong** [comment]
Yep, just opened it up for review -- seems to be working well in our prod environment.

File: types/gapi.client.classroom/index.d.ts
Comment (andrewbranch) [inline types/gapi.client.classroom/index.d.ts]: What in these definitions makes 3.7 the lowest compatible TypeScript version?

**Comment by andrewbranch** [comment]
Thanks for adopting this @declanvong! Is it ready to merge after you fix the minimum TS version?

**Comment by elibarzilay** [comment]
ping?

**Comment by declanvong** [comment]
We're still actively fixing some minor bugs first over in https://github.com/Maxim-Mazurok/google-api-typings-generator. I think the last blocking change was merged last week so I'll regenerate and test these on my project one last time, and re-request review then.

**Comment by weswigham** [comment]
... ping?

**Comment by declanvong** [comment]
Updated, sorry for the wait :) 

cc @Maxim-Mazurok

**Comment by jessetrinity** [comment]
Still waiting on review?

**Comment by Maxim-Mazurok** [comment]
Yeah, it may take some time since the last author of typings went offline, so TS team reviewers will review it eventually. Probably 1-2 weeks tops.

File: types/gapi.client.classroom/tsconfig.json
Comment (jessetrinity) [inline types/gapi.client.classroom/tsconfig.json]: trailing whitespace?

File: types/gapi.client.classroom/index.d.ts
Comment (jessetrinity) [inline types/gapi.client.classroom/index.d.ts]: I don't usually like to see authors removed except by themselves. Does @Bolisov no longer want to see updates?

**Review by jessetrinity — CHANGES_REQUESTED** [review]

**Comment by jessetrinity** [comment]
Awesome. I don't have much to add except the comment about the owner.

File: types/gapi.client.classroom/index.d.ts
Comment (declanvong) [reply types/gapi.client.classroom/index.d.ts]: They were pinged much earlier on by TS bot here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40407#issuecomment-554205819

They were the original authors of the type generator which we've forked since they've gone unresponsive. Their profile also indicates that they haven't had any Github activity [since 2018](https://github.com/Bolisov) ¯\\\_(ツ)_/¯

File: types/gapi.client.classroom/tsconfig.json
Comment (declanvong) [reply types/gapi.client.classroom/tsconfig.json]: Oops, good catch, thanks!

```suggestion
        "forceConsistentCasingInFileNames": true,
```

**Comment by Maxim-Mazurok** [comment]
The initial author @Bolisov wasn't active for the last few years, so we've made fork of his generator and now improving it and trying to automate typings updates. So, to streamline the process of updating and reviewing, we are changing the author.
We already did it with Google maps typings and I've seen massive increase in downloads on NPM since the update.

**Pull request merged**
